_posts/2018-02-28-smart-window-shades.md

```diff
@@ -0,0 +1,33 @@
+
+The next part of this project was controlling the motor from a website. I started by reading about **Flask** and making a quick Hello world application on the Pi's IP. Since Flask uses python, and so does my code for the shades, I was simply able to combine the two. I learned about **HTML** and **bootstrap** while creating the index page for this application.
+
+I now needed a domain that people could type in, because remembering an IP address is only possible when you type it in countless times as the developer. Thankfully the RTPs helped me out during the OpCommathon by showing me OpenShift. As soon as I ran it up, it crashed. Rpi.GPIO imports can only be run on a Raspberry Pi... I eventually decided to make a second Flask app that would simply run on shades.csh.rit.edu and link to the Pi's IP.
+
+This was not ideal because every time you changed the height of the shade the page would reload, which does not look pretty, and the link was to an IP address. So upon suggestions I decided to implement **Ajax**.
+
+I had never used JavaScript, nor Ajax prior to this, so it was a bit of learning curve. Obstacle one was making the script work, took a while to realize that script needed to be called after the buttons were created. Then came other obstacles, such as not being able to POST to PI because it ran on HTTP and the website ran on HTTPS. Somewhere in there I had to implement CORS or Cross-Origin Resource Sharing, else it would crash the program. I tried for the longest time to have HTTPS with the PI's IP, but it never worked perfectly, so I decided to go with allowing HTTP requests on the OpenShift/CSH domain.
+
+I proceeded to add some more functionality to this application, which took a days worth of work because I was still learning JavaScript. Now you can click buttons to set the height in percent, change the motor steps needed for 0% to 100%, and move the motor n steps at a time for debugging purposes. Furthermore, it  shows the current shade percentage, so if it does not match up with what you have then you can know that you need change the numbers up.
```

Comment:
* `s/days/day's/`
* `s/height in percent/height as a percentage/`
* Clarify "change the motor steps needed for 0% to 100%"
* `s/you need change/you need to change/`